x-pack/plugins/transform/public/app/common/request.ts

```diff
@@ -130,48 +130,60 @@ export function getPreviewTransformRequestBody(
   return request;
 }
 
+export const getCreateTransformSettingsRequestBody = (
+  transformDetailsState: StepDetailsExposedState
+): { settings: PutTransformsRequestSchema['settings'] } => {
+  // have to use Record<> here because typeof schema is readonly
+  const tempSettings = {} as Record<string, number | null | undefined>;
+
+  if (transformDetailsState.transformSettingsMaxPageSearchSize) {
+    tempSettings.max_page_search_size = transformDetailsState.transformSettingsMaxPageSearchSize;
+  }
+  if (transformDetailsState.transformSettingsDocsPerSecond) {
+    tempSettings.docs_per_second = transformDetailsState.transformSettingsDocsPerSecond;
+  }
+
+  return { settings: tempSettings };
+};
```

Comment:
If we repeat the ternary operator pattern on this inner function we can construct the `settings` object to satisfy the existing types with the `readonly` setting:

```suggestion
export const getCreateTransformSettingsRequestBody = (
  transformDetailsState: StepDetailsExposedState
): { settings?: PutTransformsRequestSchema['settings'] } => {
  const settings: PutTransformsRequestSchema['settings'] = {
    ...(transformDetailsState.transformSettingsMaxPageSearchSize
      ? { max_page_search_size: transformDetailsState.transformSettingsMaxPageSearchSize }
      : {}),
    ...(transformDetailsState.transformSettingsDocsPerSecond
      ? { docs_per_second: transformDetailsState.transformSettingsDocsPerSecond }
      : {}),
  };

  return Object.keys(settings).length > 0 ? { settings } : {};
};
```

Reply:
Updated here https://github.com/elastic/kibana/pull/78364/commits/665232fd43a76aa120784f80400a2878f383a08b